crates/cli/src/subcommands/server.rs

```diff
@@ -379,6 +379,7 @@ pub async fn exec_edit(mut config: Config, args: &ArgMatches) -> Result<(), anyh
     }
 
     let (old_nick, old_host, old_proto) = config.edit_server(server, new_nick, new_host, new_proto)?;
+    let server = new_nick.unwrap_or(server);
```

Comment:
this causes the value of `server` to be correct for any code below (i.e. any code after `config.edit_server`).